From 682f60c7bdf30c7229a795fed0260e2dabff736e Mon Sep 17 00:00:00 2001
From: Liu Ying <victor.liu@nxp.com>
Date: Fri, 13 Oct 2017 11:47:51 +0800
Subject: [PATCH 2641/5242] MLK-16581-2 phy: mixel-lvds-combo: Get PHY clock
 rate before setting it's rate

commit  a97435d2920bc1d2a1d21d1a5be339411eeabff8 from
https://source.codeaurora.org/external/imx/linux-imx.git

Due to i.MX8 clock issue, we need to get PHY clock rate
before setting it's rate when system resumes back from
PM sleep mode, otherwise, we'll fail to set the clock rate.
So, this is a workaround and it can be removed when
the clock issue is properly fixed.

Signed-off-by: Liu Ying <victor.liu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/phy/phy-mixel-lvds-combo.c |    7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/drivers/phy/phy-mixel-lvds-combo.c b/drivers/phy/phy-mixel-lvds-combo.c
index 9d3d469..3b888b1 100644
--- a/drivers/phy/phy-mixel-lvds-combo.c
+++ b/drivers/phy/phy-mixel-lvds-combo.c
@@ -96,6 +96,13 @@ void mixel_phy_combo_lvds_set_phy_speed(struct phy *phy,
 {
 	struct mixel_lvds_phy *lvds_phy = phy_get_drvdata(phy);
 
+	/*
+	 * To workaround setting clock rate failure issue
+	 * when the system resumes back from PM sleep mode,
+	 * we need to get the clock rate before setting it's
+	 * rate, otherwise, setting the clock rate will fail.
+	 */
+	clk_get_rate(lvds_phy->phy_clk);
 	clk_set_rate(lvds_phy->phy_clk, phy_clk_rate);
 }
 EXPORT_SYMBOL_GPL(mixel_phy_combo_lvds_set_phy_speed);
-- 
1.7.9.5

